dallinger/prolific.py

```diff
@@ -146,20 +201,63 @@ def draft_study(
     ) -> dict:
         """Create a draft Study on Prolific, and return its properties."""
 
+        config = get_config()
+        config.load()
+
+        # Ensure the workspace specified in the configuration exists, and get its id.  Configuration.get() will raise an
+        # exception if the key doesn't exist.
+        workspace_name = config.get("prolific_workspace")
+
+        try:
+            workspace_id = self._translate_workspace_name(workspace_name)
+
+        except ProlificServiceNoSuchWorkspace:
+            # The user specified a workspace but it doesn't exist.  Create it.
```

Comment:
I'm not sure it's a good idea to create workspaces automatically if they don't exist. Each workspace has its own finances, so if you deploy an experiment and the workspace is created on deployment then it won't have any money and you won't be able to recruit any participants. I think instead we should check (prior to deployment, on the local machine) whether the workspace exists, and if it doesn't exist, instruct the user to either create it or change the config variable to select the desired workspace.

cc. @jacobyn

Reply:
Got it.  I'll change it to raising an exception with a suitable message.

Reply:
I just pushed up changes reflecting your feedback.

Reply:
I am on board with Peter's suggestion. I still think that Projects should be created in the same way as before (created by default if does not exist)